plugins/CuraDrive/src/RestoreBackupJob.py

```diff
@@ -60,32 +71,107 @@ def _onRestoreRequestCompleted(self, reply: QNetworkReply, error: Optional["QNet
+            self._job_done.set()
+            return
 
         # Tell Cura to place the backup back in the user data folder.
-        with open(temporary_backup_file.name, "rb") as read_backup:
+        metadata = self._backup.get("metadata", {})
+        with open(self._temporary_backup_file.name, "rb") as read_backup:
             cura_api = CuraApplication.getInstance().getCuraAPI()
-            cura_api.backups.restoreBackup(read_backup.read(), self._backup.get("metadata", {}))
+            cura_api.backups.restoreBackup(read_backup.read(), metadata, auto_close=False)
+
+        # Read packages data-file, to get the 'to_install' plugin-ids.
+        version_to_restore = Version(metadata.get("cura_release", "dev"))
+        version_str = f"{version_to_restore.getMajor()}.{version_to_restore.getMinor()}"
+        packages_path = os.path.abspath(os.path.join(os.path.abspath(
+            Resources.getConfigStoragePath()), "..", version_str, "packages.json"))
+        if not os.path.exists(packages_path):
+            Logger.error(f"Can't find path '{packages_path}' to tell what packages should be redownloaded.")
+            self.restore_backup_error_message = self.DEFAULT_ERROR_MESSAGE
+            self._job_done.set()
+            return
+
+        to_install = {}
+        try:
+            with open(packages_path, "r") as packages_file:
+                packages_json = json.load(packages_file)
+                if "to_install" in packages_json:
+                    for package_data in packages_json["to_install"].values():
+                        if "package_info" not in package_data:
+                            continue
+                        package_info = package_data["package_info"]
+                        if "package_id" in package_info and "sdk_version_semver" in package_info:
+                            to_install[package_info["package_id"]] = package_info["sdk_version_semver"]
+        except IOError as ex:
+            Logger.error(f"Couldn't open '{packages_path}' because '{str(ex)}' to get packages to re-install.")
+            self.restore_backup_error_message = self.DEFAULT_ERROR_MESSAGE
+            self._job_done.set()
+            return
+
+        if len(to_install) < 1:
+            Logger.info("No packages to reinstall, early out.")
+            self._job_done.set()
+            return
 
-        self._job_done.set()
+        # Download all re-installable plugins packages, so they can be put back on start-up.
+        redownload_errors = []
+        def packageDownloadCallback(package_id: str, msg: "QNetworkReply", err: "QNetworkReply.NetworkError" = None) -> None:
+            if err is not None or HttpRequestManager.safeHttpStatus(msg) != 200:
+                redownload_errors.append(err)
+            del to_install[package_id]
+
+            try:
+                with NamedTemporaryFile(mode="wb", suffix=".curapackage", delete=False) as temp_file:
```

Comment:
You can probably use `delete_on_close` now that we have Python 3.12 :slightly_smiling_face: